### PR TITLE
Fix config import feature

### PR DESCRIPTION
**Pull request by blms**
Requesting @tashrifbillah review for only the (minor) bash script and README changes.

### Changes
- Fix the feature to import configurations from the web interface upload button
  - Replace the original CSV upload-and-process code with streamlined JSON parsing, which does not require writing to the filesystem
  - Add parsing JSON with schema supplied to `yup` schema validator
- Remove deprecated code from bash scripts
- Remove `build.sh` which would now be identical to running `npm run build`
- Add related information to `README-PNL.md`

For now, we only accept JSON configurations. We could accept CSV in the future, but for the sake of getting *some* kind of configuration upload working per user request, this is it for now.

### Testing
- [x] Deleted `${containerDataDir}/dpdash/uploads` directory to simulate re-running `init.sh` with the changes
- [x] Rebuilt all code
- [x] Rebuilt image `/data/predict/dpdash-ben-0611.sif`
- [x] Tested all affected functionality

### CODE REVIEW

**Comment by tashrifbillah** [comment]
> Rebuilt image /data/predict/dpdash-ben-0611.sif

Are we pushing up the new image to Dropbox once in a while? If not, we should.

File: README-PNL.md [hunk 1]
```diff
@@ -216,7 +216,11 @@ in the remote server to be able to view the login page in a local web browser.
 
 ### Configure
 
-Add and enable at least one configuration from the left taskbar.
+Add and enable at least one configuration by selecting "Configure" from the left taskbar. 
+
+You may create a new configuration by clicking the "Add a configuration manually" button in the bottom-right corner of the Configuration page.
+
+Alternatively, to upload a configuration from a JSON file, you may use the "Uplaod configuration file" button in the bottom-right corner of the Configuration page. The JSON file you upload must match the [schema detailed here](https://github.com/PREDICT-DPACC/dpdash/wiki/Configuration-schema).
```
Comment (tashrifbillah) [inline README-PNL.md:223]: Just for the record--the suggested JSON schema looks like:

image

Comment (blms) [reply README-PNL.md:223]: Yes.

**Comment by tashrifbillah** [comment]
> Remove build.sh which would now be identical to running npm run build

Was `build.sh` used anywhere? I can't find any usage in `init.sh`. Regardless, when should a developer run `npm run build`? We do that only in the Singularity recipe [here](https://github.com/PREDICT-DPACC/dpdash/blob/pnl-devel/singularity/Singularity#L74).

**Comment by blms** [comment]
> > Rebuilt image /data/predict/dpdash-ben-0611.sif
> 
> Are we pushing up the new image to Dropbox once in a while? If not, we should.

I haven't been but definitely can do that, thanks for the reminder.

File: README-PNL.md (hunk 1 above)
Comment (tashrifbillah) [reply README-PNL.md:223]: If appropriate, please consider adding such an indented view in the wiki. I obtained the above from http://jsonviewer.stack.hu/

File: singularity/push.sh [hunk 2]
```diff
@@ -16,9 +16,6 @@ export RABBITMQ_LOG_BASE=/data/dpdash/rabbitmq
 # vars for celery worker
 export dppy_config=/data/dpdash/configs/dppy.conf
 
-# vars for DPdash node app
```
Comment (tashrifbillah) [inline singularity/push.sh:19]: I am sure you detected it right but can you explain why the two variables were here before and why don't we need them anymore? Zoom call is okay if that would be easier ...

**Comment by blms** [comment]
> > Remove build.sh which would now be identical to running npm run build
> 
> Was `build.sh` used anywhere? I can't find any usage in `init.sh`. Regardless, when should a developer run `npm run build`? We do that only in the Singularity recipe [here](https://github.com/PREDICT-DPACC/dpdash/blob/pnl-devel/singularity/Singularity#L74).

It was not used in any other script. It was only meant for those (like me) who are actively developing features—`npm run build` is only useful during development when using the `-d` flag for `start.sh`. In other words, when a development directory is mounted. Then, it will rebuild all code into `/public/js/*.min.js` and `/dist`. Developers may use this command to compile changes they have made to the JS code in the mounted development directory.

If run without a development directory mounted, it will not accomplish anything, since it will compile the code that is already on the container, which was already compiled during the Singularity recipe.

**Review by tashrifbillah — APPROVED** [review]
Config import functionality was disabled for a long time. It's good that this PR is restoring that. It can be merged upon providing a few clarifying comments. Thanks.

File: singularity/push.sh (hunk 2 above)
Comment (blms) [reply singularity/push.sh:19]: They were used only by the old "configuration import" feature, and my replacement code doesn't need them.

`DPDASH_UPLOADS` was an upload directory for Node.JS to write to the filesystem. The old feature saved the CSV to that directory before processing it. My code just takes in JSON and processes it in-memory.

`DPDASH_UPLOADS_CONFIG_SCHEMA` has been deprecated by my function `getConfigSchema`. My reasoning for moving this into code is that the needed fields for a configuration are already hard-coded, so a schema to match that can also live in code alongside that. If the schema changes, then the whole architecture of configurations would also need to change.

File: README-PNL.md (hunk 1 above)
Comment (blms) [reply README-PNL.md:223]: Sure, I'll just include your image! Certainly can't hurt.

**Pull request merged**